test/e2e/json-rpc/eth_newBlockFilter.spec.js

```diff
@@ -0,0 +1,88 @@
+const { strict: assert } = require('assert');
+const { convertToHexValue, withFixtures } = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+describe('eth_newBlockFilter', function () {
+  const ganacheOptions = {
+    blockTime: 0.1,
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
+  it('executes a new block filter call', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // eth_newBlockFilter
+        await driver.openNewPage(`http://127.0.0.1:8080`);
+
+        const newBlockfilterRequest = JSON.stringify({
+          jsonrpc: '2.0',
+          method: 'eth_newBlockFilter',
+        });
+
+        const newBlockFilter = await driver.executeScript(
+          `return window.ethereum.request(${newBlockfilterRequest})`,
+        );
+
+        assert.strictEqual(newBlockFilter, '0x01');
+
+        // eth_getFilterChanges
+        const getFilterChangesRequest = JSON.stringify({
+          jsonrpc: '2.0',
+          method: 'eth_getFilterChanges',
+          params: ['0x01'],
+        });
+
+        await driver.delay(1000);
```

Comment:
could we possibly skip this delay and wait for a condition instead?

Reply:
How blocking is this? We need to delay to move to a new block for the filter to respond. We can get the log output and wait for that?

Reply:
hey @tmashuang it's not a blocking issue :) just wondering there was a way to avoid using the delay but that's fine as it is :)